tests/blackbox.py

```diff
@@ -440,3 +477,425 @@ def dryrun_one(
+                approve_sim = Simulation(
+                    self.algod,
+                    ExecutionMode.Application,
+                    approval_teal,
+                    self.predicates[meth_name],
+                    abi_method_signature=sig,
+                    identities_teal=model_approval_teal,
+                    validation=executor_validation,
+                )
+
+                for oc_str, call_cfg in asdict(meth_cfg).items():
+                    oc = as_on_complete(oc_str)
+
+                    def simulate_approval(on_create):
+                        tp: TxParams = deepcopy(txn_params)
+                        tp.update_fields(
+                            TxParams.for_app(is_app_create=on_create, on_complete=oc)
+                        )
+                        sim_results = approve_sim.run_and_assert(
+                            approval_strat, txn_params=tp, msg=msg4simulate()
+                        )
+                        assert sim_results.succeeded
+                        if meth_name not in self.results:
+                            self.results[meth_name] = {}
+                        self.results[meth_name][(on_create, oc)] = sim_results
+                        update_stats(meth_name, len(self.predicates[meth_name]))
+
+                    # weird walrus is_app_create := ... to fill closure of msg4simulate()
+                    if call_cfg & CallConfig.CALL:
+                        double_check_at_least_one_method = True
+                        simulate_approval(is_app_create := False)
+
+                    if call_cfg & CallConfig.CREATE:
+                        double_check_at_least_one_method = True
+                        simulate_approval(is_app_create := True)
+            assert double_check_at_least_one_method, "no method was simulated"
+
+        # ---- CLEAR PROGRAM SIMULATION ---- #
+        clear_strat_or_inputs: InputStrategy  # CallStrategy | Iterable[Sequence[PyTypes]]
+        clear_sim: Simulation | None = None
+        if not omit_clear_call:
+            assert clear_args_strat_type_or_inputs  # therefore Type[ABIStrategy] | list[Sequence[PyTypes]]
+            if isinstance(clear_args_strat_type_or_inputs, list):
+                clear_strat_or_inputs = cast(
+                    list[Sequence[PyTypes]], clear_args_strat_type_or_inputs
+                )
+                # for the closure of local update_stats():
+                num_dryruns = len(clear_strat_or_inputs)
+            else:
+                clear_strat_or_inputs = RandomArgLengthCallStrategy(
+                    cast(Type[ABIStrategy], clear_args_strat_type_or_inputs),
+                    max_args=2,
+                    num_dryruns=num_dryruns,
+                    min_args=0,
+                    type_for_args=sdk_abi.ABIType.from_string("byte[8]"),
+                )
+
+            meth_name = CLEAR_STATE_CALL
+            is_app_create = False
```

Comment:
same as previous comment on asking for putting declaration of `is_app_create` ahead of testing for both apporval app calls and clear state app calls.

Reply:
see https://github.com/algorand/pyteal/pull/634/commits/deca0acba00c971c10dbe1e7c7a78a60cf7c641d

Reply:
thanks! looks good